### DISCUSSION TITLE
New Postgres Storage Size Increasing...Exponentially?

### DISCUSSION THREAD

**Discussion by ckhicks** [vercel/community#2981] [Help] [answered]
### Summary

I created a fresh DB two days ago and have been building out some schema using TablePlus. I have added maybe one or two records but am seeing storage sizes of over 600 MB as of today...and it's quickly rising? 😅 Attached a couple of screenshots for reference.

Anyone else seen anything like this?

<img width="1240" alt="Screen Shot 2023-06-23 at 3 51 35 PM" src="https://github.com/vercel/community/assets/767012/820a5947-3163-4a42-9cea-8c5debba0481">
<img width="946" alt="Screen Shot 2023-06-23 at 3 59 56 PM" src="https://github.com/vercel/community/assets/767012/f004c8bd-e3ab-4f57-8602-ea02f4232038">

### Steps to Reproduce

The connected application is a brand new Next.js app with almost nothing in it. I'm using Kysely for reads - which worked swimmingly - but have no idea why my DB is exploding in size with almost no data written.

**Comment by ckhicks** [comment]
Update: I am suddenly seeing a massive change in storage...but still nowhere near the actual _one row_ of data that I have in here. 🙃
 
 ![Screen Shot 2023-06-23 at 8 05 56 PM](https://github.com/vercel/community/assets/767012/224997ab-6007-4644-b4bc-fbb151f91ba9)

**Reply by ckhicks** [reply]
...now back over 700MB, no data changes. 😂

**Comment by wzpfish** [comment]
![image](https://github.com/vercel/community/assets/9511595/d936de7b-f373-4a2a-a179-945b30b761b0)
The same, there is very little data in my postgres database (less then 100 records),  but the storage suddenly increase to more than 400MB...

**Comment by z4nr34l** [comment]
Same here, empty DB is over limited size
<img width="184" alt="image" src="https://github.com/vercel/community/assets/5320734/a74df14e-f371-45c0-a8f3-1c1b20a475ec">

**Comment by duck-goes-quack** [comment]
oh I see this isn't just a me problem... has this always been the case or is this a new thing?

![Capture](https://github.com/vercel/community/assets/112182146/a80630fa-2f7c-4ba5-b8d7-1ec11f786d1d)

**Reply by duck-goes-quack** [reply]
Because my database of 100 rows is like 70mb apparently. But just adjusting one row element (not even a new record just updating one) caused it to go up by like 200mb ! And three requests in trying to figure out why there was an error I see this in my storage

**Reply by errorcodezero** [reply]
I put in I think 1 row and I have around the same amount.

**Comment by e-hart** [comment]
Glad to see I'm not the only one 😅

**Reply by duck-goes-quack** [reply]
right... I hope they fix it soon. This is my first time on this forum, is this main place of discussion or is there other places like a discord?

I need to do an update for my project website in ~1 week so gonna have to get a bandaid local cache thing to do for now... however I got to do a merchandise upgrade in ~2 weeks, I definitely need to have ability to have like at least 100-200 rows of data written... whereas right now its like 2-3 and it goes over...

**Comment by errorcodezero** [comment]
Same for me.

**Comment by Jefishub** [comment]
Same here. Made new database today aand... 
![image](https://github.com/vercel/community/assets/69782705/e4f95705-6bed-41c3-a23d-9c5fc6460e59)

**Comment by aklevecz** [comment]
Also came here for this. I'm assuming it's a UI bug or a query gone haywire
<img width="663" alt="Screenshot 2023-06-24 at 5 04 31 PM" src="https://github.com/vercel/community/assets/38079642/c22b02b4-ad82-4930-9c3e-bc9d202fb1f7">

**Comment by penguinboots** [comment]
![image](https://github.com/vercel/community/assets/60584906/9ede7078-f26c-40b4-98a3-a14d4fba8bde)

This had me extremely confused! It's just a handful of tables for a tiny project which never had more than 10MB of use on top of the 31MB for Postgres itself..

**Reply by JeffRice** [reply]
Looks like it is changing actively. My usage is still in the red, but the numbers are actually going down for storage size and compute time

**Comment by da-stoi** [comment]
Same issue here. I have under 20 rows across all tables
<img width="325" alt="image" src="https://github.com/vercel/community/assets/20031472/47320e72-10cd-4f1f-8b99-d96df3a93b75">

**Comment by amyegan** [comment, marked as answer]
Hi everyone! We had a display bug for storage size reporting which is fixed now. You should no longer see this issue, so please let me know if it is still happening for you 🙇

**Reply by ckhicks** [reply]
Thank you very much!

**Reply by errorcodezero** [reply]
👍🏾

**Reply by penguinboots** [reply]
Thanks!

**Reply by bmmathe** [reply]
I created a database two days ago, put about 60 small records in it and I'm showing > 150MB of storage usage and only ~300kB of transfer.  Something is still very wrong.

**Reply by amyegan** [reply]
Hey @bmmathe. Would you mind opening a new discussion? That will make it easier track as this is a similar but new issue. If you include the project name or domain then I can locate the project to investigate.

**Reply by JolanPoussier** [reply]
Hi @amyegan, i have the same issue on my DB. It's still rising even though I haven't put a lot in it and nothing in a while.
The project name is : project-rl-database

**Reply by amyegan** [reply]
@JolanPoussier Thanks for sharing the name. It looks like there was about 40MB of written data on October 3 with no change in storage size since then. Can you share where you see the size still rising? That will help us investigate the right place 🙏 

@bmmathe It looks like more data was written to the database that was read out of it. For clarification, these are the relevant usage numbers to consider:
- [Storage Size](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#storage) is the maximum amount of storage used per month, including the 33MB of space used by Postgres itself. 
- [Written data](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#written-data) is the amount of data written from your functions **to** your databases. 
- [Data transfer](https://vercel.com/docs/storage/vercel-postgres/usage-and-pricing#data-transfer) is the amount of data returned to your functions **from** your database.

**Reply by JolanPoussier** [reply]
@amyegan Thank you for the quick answer ! The problem is still at the same place as the others; in the Storage Size
<img width="681" alt="Capture d’écran 2023-11-23 à 10 07 16" src="https://github.com/vercel/community/assets/113784373/49c8b906-7212-47d9-a5e8-10e8836b5855">

**Reply by JolanPoussier** [reply]
@amyegan Hi again, the storage isn't increasing anymore but the datas are still at 214, is it normal ? Can I had more without this number to be a problem ?
